eth/backend.go

```diff
@@ -559,6 +576,75 @@ func (self *Ethereum) syncAccounts(tx *types.Transaction) {
 	}
 }
 
+func (self *Ethereum) StartAutoDAG() {
+	go func() {
+		glog.V(logger.Info).Infof("Automatic pregeneration of ethash DAG ON (ethash dir: %s)", self.AutoDAGDir)
+		var nextEpoch uint64
+		ticker := time.NewTicker(autoDAGcheckInterval)
+		self.autodagquit = make(chan bool)
+		for {
+			select {
+			case <-ticker.C:
```

Comment:
I first did it like that, but concluded this was the better solution. You can only miss epochs if you are not properly synced and we advance with epochs via download, then ethash just falls back to on-demand generation.